### PR TITLE
fixed UserJWTPayload property typo

### PR DESCRIPTION
**Pull request by CowboyBebop**

### CODE REVIEW

**Comment by Pierre-Demessence** [comment]
What's this PR for ? Where is the typo ? Also your PR doesn't compile

**Comment by CowboyBebop** [comment]
The typo is that the token stores as `{ email: "email", username: "username"}` but UserJWTPayload interface is `{ id: "id", username: "username" }`. Which causes an error in `populateUser` as `user` will always be null since `id` will always be null as well.

The build didn't trigger again for some reason.

**Comment by Pierre-Demessence** [comment]
> The typo is that the token stores as `{ email: "email", username: "username"}` but UserJWTPayload interface is `{ id: "id", username: "username" }`. Which causes an error in `populateUser` as `user` will always be null since `id` will always be null as well.
> 
> The build didn't trigger again for some reason.

 The build did trigger, and the fix should be rather be here: https://github.com/Corniland/server/blob/master/src/routes/auth/user.ts#L79

`const accessToken = jwt.sign(userEmailDoc.getJWTPaylod(), process.env.JWT_SECRET_USER);`

**Comment by CowboyBebop** [comment]
Actually yes, but in both places, it would be better to add id to the stored token data and to the email to the interface.
Also, yes i rushed too fast rn

**Comment by Pierre-Demessence** [comment]
You're the one who removed the ID from the payload...

**Comment by CowboyBebop** [comment]
I did, and i also realized the better solution is to add both data.

**Comment by Pierre-Demessence** [comment]
But why do you need the email in the JWT Payload?

**Comment by CowboyBebop** [comment]
Technically it's not needed, but it could be viably needed for something else later.

**Comment by CowboyBebop** [comment]
In fact even the `username` is not needed right now, it's just that id, email and username is the standard data you could expect in the token

File: src/middleware/auth/user.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const populateUser: express.RequestHandler = compose([
     try {
       if (req.token) {
         const payload = getUserJWTPayload(req.token);
-        const user = await UserModel.findById(payload?.id);
+        const user = await UserModel.findOne({ username: payload?.username });
```
Comment (Pierre-Demessence) [inline src/middleware/auth/user.ts]: To be reverted

**Review by Pierre-Demessence — CHANGES_REQUESTED** [review]

**Pull request merged**